applications/OptimizationApplication/custom_utilities/container_variable_data_holder/container_variable_data_holder.h

```diff
@@ -0,0 +1,157 @@
+    ContainerVariableDataHolder(ModelPart& rModelPart)
+        : BaseType(rModelPart)
+    {
+    }
+
+    /// Copy constructor
+    ContainerVariableDataHolder(const ContainerVariableDataHolder& rOther)
+        : BaseType(rOther)
+    {
+    }
+
+    /// Copy constructor with base class used to transfer data between compatible data containers
+    ContainerVariableDataHolder(const BaseType& rOther)
+        : BaseType(rOther)
+    {
+    }
+
+    ~ContainerVariableDataHolder() override = default;
+
+    ///@}
+    ///@name Public operations
+    ///@{
+
+    ContainerVariableDataHolder::Pointer Clone() const;
+
+    ContainerVariableDataHolder::Pointer CloneWithDataInitializedToZero() const;
+
+    template <class TDataType>
+    void ReadDataFromContainerVariable(
+        const Variable<TDataType>& rVariable);
+
+    template <class TDataType>
+    void AssignDataToContainerVariable(
+        const Variable<TDataType>& rVariable);
+
+    template <class TDataType>
+    void SetDataForContainerVariable(
+        const Variable<TDataType>& rVariable,
+        const TDataType& rValue);
+
+    template <class TDataType>
+    void SetDataForContainerVariableToZero(
+        const Variable<TDataType>& rVariable);
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ContainerVariableDataHolder operator+(const ContainerVariableDataHolder& rOther) const;
+
+    ContainerVariableDataHolder& operator+=(const ContainerVariableDataHolder& rOther);
+
+    ContainerVariableDataHolder operator+(const double Value) const;
+
+    ContainerVariableDataHolder& operator+=(const double Value);
+
+    ContainerVariableDataHolder operator-(const ContainerVariableDataHolder& rOther) const;
+
+    ContainerVariableDataHolder& operator-=(const ContainerVariableDataHolder& rOther);
+
+    ContainerVariableDataHolder operator-(const double Value) const;
+
+    ContainerVariableDataHolder& operator-=(const double Value);
+
+    ContainerVariableDataHolder operator*(const ContainerVariableDataHolder& rOther) const;
+
+    ContainerVariableDataHolder& operator*=(const ContainerVariableDataHolder& rOther);
+
+    ContainerVariableDataHolder operator*(const double Value) const;
+
+    ContainerVariableDataHolder& operator*=(const double Value);
+
+    ContainerVariableDataHolder operator/(const ContainerVariableDataHolder& rOther) const;
+
+    ContainerVariableDataHolder& operator/=(const ContainerVariableDataHolder& rOther);
+
+    ContainerVariableDataHolder operator/(const double Value) const;
+
+    ContainerVariableDataHolder& operator/=(const double Value);
+
+    ContainerVariableDataHolder operator^(const ContainerVariableDataHolder& rOther) const;
+
+    ContainerVariableDataHolder& operator^=(const ContainerVariableDataHolder& rOther);
+
+    ContainerVariableDataHolder operator^(const double Value) const;
+
+    ContainerVariableDataHolder& operator^=(const double Value);
+
+    ContainerVariableDataHolder& operator=(const ContainerVariableDataHolder& rOther);
```

Comment:
I'd put this somewhere near the constructors instead of the operators.

Reply:
moved